File: src/image_converter.go
package main

import (
	"fmt"
	"image"
	"image/jpeg"
	"log"
	"os"
	"path/filepath"

	_ "image/png"
)

// convertImageToJPG конвертирует изображение в формат JPEG.
func convertImageToJPG(inputPath string, outputPath string) error {
	// Открываем входной файл
	inputFile, err := os.Open(inputPath)
	if err != nil {
		return err
	}
	defer inputFile.Close()

	// Декодируем изображение
	img, _, err := image.Decode(inputFile)
	if err != nil {
		return err
	}

	// Создаем выходной файл
	outputFile, err := os.Create(outputPath)
	if err != nil {
		return err
	}
	defer outputFile.Close()

	// Кодируем изображение в формате JPEG
	if err := jpeg.Encode(outputFile, img, nil); err != nil {
		return err
	}

	return nil
}

func main() {
	if len(os.Args) < 3 {
		fmt.Println("Usage: go run main.go <input_image> <output_image>")
		return
	}

	inputPath := os.Args[1]
	outputPath := os.Args[2]

	// Проверяем расширение выходного файла
	if filepath.Ext(outputPath) != ".jpg" && filepath.Ext(outputPath) != ".jpeg" {
		log.Fatalf("Output file must have .jpg or .jpeg extension")
	}

	// Конвертируем изображение
	if err := convertImageToJPG(inputPath, outputPath); err != nil {
		log.Fatalf("Failed to convert image: %v", err)
	}

	if _, err := os.Stat(outputPath); os.IsNotExist(err) {
    log.Fatalf("Output file not created: %v", outputPath)
}

	fmt.Printf("Successfully converted %s to %s\n", inputPath, outputPath)
}

File: src/image_converter.cpp
#define cimg_display 0  // Disable display if you don't have X11
#include "CImg.h"
#include <iostream>
#include <string>

using namespace cimg_library;

int main(int argc, char** argv) {
    // Check for correct number of arguments
    if (argc != 3) {
        std::cerr << "Usage: " << argv[0] << " <input_image> <output_image.jpg>" << std::endl;
        return 1;
    }

    std::string input_image = argv[1];     // Input image file
    std::string output_image = argv[2];    // Output image file (should be .jpg)

    try {
        // Load the input image
        CImg<unsigned char> img(input_image.c_str());

        // Save the image as JPEG
        img.save(output_image.c_str());

        std::cout << "Image converted successfully: " << output_image << std::endl;
    } catch (CImgException& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}